src/fileio.c

```diff
@@ -135,13 +135,11 @@ mame_file *mame_fopen(const char *gamename, const char *filename, int filetype,
 
 		/* high score files */
 		case FILETYPE_HIGHSCORE:
-			if (!mame_highscore_enabled())
-				return NULL;
 			return generic_fopen(filetype, NULL, gamename, 0, openforwrite ? FILEFLAG_OPENWRITE : FILEFLAG_OPENREAD);

```

Comment:
i don’t know why we remove stuff like this. even if we always set hiscores on, it can’t hurt to leave the hooks to the option in, in case we need to disable them later for whatever reason.

Reply:
My approach for my mame2003-plus contributions is to remove things like this completely.

I guess mame2003-libretro might want to make it possible to disable using samples or saving hiscores etc for some reason down the line. To me, mame_hiscore_enabled() isn't even the way we would want to do that -- I'd prefer to code a new hook for disabling them that is cleaner and not meant to "protect" my hiscore table from saving scores when I have cheats enabled like this implementation.

Anyway I'm not going to go back and re-add the old hooks for disabling samples, cheats, or hiscores etc. to mame2003-libretro but if there is consensus that old hooks should stay I'll avoid commenting them out in the future when porting things over.

Reply:
its already changed well i dont know who looks after this core i assumed it was you @dankcushions

Reply:
im a bit lost in your comments that was my original stance that we should get the core working like the original then move on to libretto. But we cant do that if the mame gui is going this has to be done and its been said that this is libretro core and by that definition it has to be ripped apart and changed you cant say leave mame core alone and it shouldnt be a core just running in libretro

Reply:
removing unused code and refactoring is not a precursor to replacing the MAME osd. if i was hypothetically going to do that i'd probably want to redo the whole libretro fork, as our current hooks into the osd stuff is probably a bad foundation.

anyways i'm not against refactoring, just raising concerns. the people who want to work on the code have more stake than me. i don't even use it!